### PR TITLE
Update travis configuration

### LINKED ISSUE #226: CI doesn't handle feature flags with workspace
Correctly testing feature flags in workspaces is hard. The current CI setup doesn't handle this appropriately. For some of the issues you could look at how complicated the [`futures` workspaces `.travis.yml`](https://github.com/rust-lang-nursery/futures-rs/blob/master/.travis.yml) is and its history.

I'll try and port some of the simpler ideas from the `futures` setup across here so that as more feature flags are added basic compatibility can be tested (actually exhaustively testing feature flags becomes impossible quickly because there are `O(n!)` combinations, but with just a few extra builds you can test for the most likely issues).

### PR DESCRIPTION
**Pull request by Nemo157**
## Description
 * Separate out individual build jobs for faster wall-clock testing

 * Fix clippy not actually denying warnings (excluded examples because these are currently failing and have non-trivial fixes, https://github.com/rustasync/tide/issues/230)

 * Add build job that checks --no-default-features works

 * Add build job that checks for intra-doc-resolution failures (excluded tide because of bugs in re-exports with the intra-doc feature, https://github.com/rust-lang/rust/issues/60883)

(based on top of #220, [relevant changes](https://github.com/rustasync/tide/compare/wafflespeanut:code-split..Nemo157:update-travis))

## Motivation and Context
Fixes #226, improves feedback speed on PRs, keeps code quality high

## How Has This Been Tested?
In Travis

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [CONTRIBUTING](https://github.com/rustasync/tide/blob/master/.github/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Nemo157** [comment]
I'll open issues about the two exclusions above later.

File: README.md [hunk 1]
```diff
@@ -36,7 +36,7 @@
       API Docs
     </a>
     <span> | </span>
-    <a href="https://github.com/rustasync/tide/blob/master/.github/CONTRIBUTING.md">
+    <a href="./.github/CONTRIBUTING.md">
```
Comment (yoshuawuyts) [inline README.md]: The full URL is useful so it correctly links from crates.io and other repos. But not too big of an issue (:

File: README.md [hunk 2]
```diff
@@ -71,14 +71,15 @@ fn main() -> Result<(), std::io::Error> {
...
-- [GraphQL](https://github.com/rustasync/tide/tree/master/examples/src/graphql.rs)
+- [Hello World](./examples/src/hello.rs)
+- [Messages](./examples/src/messages.rs)
+- [Body Types](./examples/src/body_types.rs)
+- [Multipart Form](./examples/src/multipart-form/mod.rs)
+- [Catch All](./examples/src/catch_all.rs)
+- [Cookies](./examples/src/cookies.rs)
+- [Default Headers](./examples/src/default_headers.rs)
+- [GraphQL](./examples/src/graphql.rs)
+- [Staticfile](./examples/src/staticfile.rs)
```
Comment (yoshuawuyts) [inline README.md]: Same for all of these links.

**Comment by fairingrey** [comment]
Let's get this merged after #220 :+1:

**Comment by fairingrey** [comment]
Conflicts will need to be resolved following #194 but it shouldn't take too long. You can overwrite `.travis.yml` with your incoming changes as they test them anyway.

File: .travis.yml [hunk 3]
```diff
@@ -1,13 +1,30 @@
...
+matrix:
+  include:
+  - name: cargo doc
+    env: [CACHE_NAME=docs]
+    script:
+    - RUSTDOCFLAGS=-Dwarnings
+      cargo doc --all --all-features --no-deps --exclude tide
+
+  - name: cargo fmt
+    cache: false
```
Comment (pickfire) [inline .travis.yml]: Why do we need to disable cache for fmt?

**Comment by yoshuawuyts** [comment]
#220 was merged, I think this should be ready for a rebase!

**Comment by Nemo157** [comment]
Rebased

**Pull request merged**